### PR TITLE
Fix Travis CI builds, update dependencies, and update list of as-yet unimplemented functions

### LINKED ISSUE #192: Security vulnerability in browserify-sign
The `4.0.0` version of `browserify-sign` depends on a version `elliptic` that has a vulnerability. You should update it too `4.2.1`.

### LINKED ISSUE #194: Error when try to use Diffie-Hellman properties
I want to use this library in my project to archive NodeJS crypto on browser.

I already downloaded the above package. In the given example it shows `createHash`. I want to use `createDiffieHellman` and `createHmac` etc. But, it rises error. Here is my code.

```
<!doctype html>
<html>
<script src='bundle.js'></script>
<body>
<h1>Welcome</h1>
</body>
<script>
      console.log(require('crypto').createDiffieHellman(139));
</script>
</html>
```

and I facing error says, **Uncaught TypeError: require(...).createDiffieHellman is not a function** in the browser console.

Even I tried to change the code on my computer _project/node_modules/crypto-browserify/example/index.html_ and also tried to copy the repository from github, but result is same.

I can't understand what happen?

### PR DESCRIPTION
**Pull request by rawr51919**
Fixes #199 
Fixes #196 
Fixes #194 
Fixes #192

### CODE REVIEW

**Review by ljharb — APPROVED** [review]
LGTM, thanks. should be a patch update

**Comment by coltongit** [comment]
Minor note @ljharb
Node 0.1x builds broke because https://github.com/crypto-browserify/publicEncrypt grabs a later version of parse-asn1 than what was originally used for the last build 3 years ago.

**Comment by ljharb** [comment]
0.1? Or 0.10?

**Comment by coltongit** [comment]
> 
> 
> 0.1? Or 0.10?

0.11/0.12

**Comment by coltongit** [comment]
@ljharb
I figured out what was erroring out the build originally for Saucelabs. Some of the ports weren't forwarded for some strange reason
```
Error: connection refused: localtunnel.me:41870 (check your firewall settings)
Error: connection refused: localtunnel.me:34242 (check your firewall settings)
Error: connection refused: localtunnel.me:35354 (check your firewall settings)
```
These are from the last build from 3 years ago

**Comment by ljharb** [comment]
gotcha, but they’ll pass now?

**Comment by coltongit** [comment]
> 
> 
> gotcha, but they’ll pass now?

If of course you have those issues fixed, then yes, this will just fine

**Comment by coltongit** [comment]
Pretty much, if the parse-asn1 issue is fixed along with those connection refused issues, this will pass Travis CI

**Comment by coltongit** [comment]
> 
> 
> gotcha, but they’ll pass now? If not, can we pin parse-asn1 so they do?

Yep! 3.1.0 iirc is the last version that worked

**Comment by coltongit** [comment]
Any progress updates on this so far? Has the Saucelabs stuff been figured out yet?

**Comment by cwc972135903** [comment]
do you how to implement the generateKeyPair

File: README.md [hunk 1]
```diff
@@ -31,6 +31,15 @@ Here is the subset that is currently implemented:
 these features from node's `crypto` are still unimplemented.
 
 * createCredentials
+* getCurves
+* randomFill
+* randomFillSync
```
Comment (markhughes) [inline README.md:36]: `randomFill` and `randomFillSync` are implemented, we can remove this from the README

File: example/bundle.js [hunk 2]
```diff
@@ -395,17 +395,16 @@ exports.createHash = function (alg) {
...
-, 'createDecipheriv'
-, 'createSign'
-, 'createVerify'
-, 'createDeffieHellman',
-, 'pbkdf2',
-, 'randomBytes' ].forEach(function (name) {
+;['createCredentials',
+  'getCurves',
+  'randomFill',
+  'randomFillSync',
```
Comment (markhughes) [inline example/bundle.js:401]: `randomFill` and `randomFillSync` are implemented, we can remove this from the error message array

**Review by markhughes — CHANGES_REQUESTED** [review]
`randomFill` and `randomFillSync` are implemented :)

**Comment by rawr51919** [comment]
this PR will return, eventually, just was cleaning out old repos, may merge this and #201 together into one PR

**Pull request closed without merging**